ksql-engine/src/main/java/io/confluent/ksql/util/KafkaTopicClientImpl.java

```diff
@@ -235,4 +247,30 @@ private void validateTopicProperties(String topic, int numPartitions, short repl
     );
   }
 
+  private static class RetryHelper<T> {
+    T executeWithRetries(Supplier<KafkaFuture<T>> supplier) throws  InterruptedException,
+                                                                    ExecutionException {
+      int retries = 0;
+      while (retries < NUM_RETRIES) {
+        try {
+          if (retries != 0) {
+            Thread.sleep(RETRY_BACKOFF_MS);
+          }
+          return supplier.get().get();
+        } catch (ExecutionException e) {
+          if (e.getCause() instanceof  RetriableException) {
+            retries++;
+            log.info("Retrying admin request due to retriable exception. Retry no: "
+                     + retries, e);
+          } else {
+            throw e;
+          }
+        }
+      }
+      throw new ExecutionException(
+          new RuntimeException("Could not complete admin operation even after "
```

Comment:
It might be good to keep a copy of the last exception and pass it in as a cause here